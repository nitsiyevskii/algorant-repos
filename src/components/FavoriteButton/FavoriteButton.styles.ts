import { createStyles } from "src/utils/createStyles";

export const useStyles = createStyles((theme) => ({
  container: {
    width: 40,
    height: 40,
    borderRadius: 20,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: theme.colors.iconButton.background,
  },
}));

