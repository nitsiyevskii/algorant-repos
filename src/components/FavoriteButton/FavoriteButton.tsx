import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useTheme } from 'src/hooks/useTheme';
import { useStyles } from './FavoriteButton.styles';

interface FavoriteButtonProps {
  isSelected: boolean;
  onPress: () => void;
  size?: number;
  disabled?: boolean;
}

export const FavoriteButton: React.FC<FavoriteButtonProps> = ({
  isSelected,
  onPress,
  size = 24,
  disabled = false,
}) => {
  const theme = useTheme();
  const styles = useStyles();

  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={disabled}
      style={[
        styles.container,
        { opacity: disabled ? 0.5 : 1 },
      ]}
      activeOpacity={0.7}
      testID="favorite-button"
    >
      <Ionicons
        name={isSelected ? 'heart' : 'heart-outline'}
        size={size}
        color={isSelected ? theme.colors.core.error : theme.colors.iconButton.icon}
      />
    </TouchableOpacity>
  );
};

