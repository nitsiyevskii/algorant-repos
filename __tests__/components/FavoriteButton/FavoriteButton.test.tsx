import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { FavoriteButton } from 'src/components/FavoriteButton/FavoriteButton';

describe('FavoriteButton', () => {
  const mockOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when not selected', () => {
    const { toJSON } = render(
      <FavoriteButton isSelected={false} onPress={mockOnPress} />
    );
    expect(toJSON()).toBeTruthy();
  });

  it('renders correctly when selected', () => {
    const { toJSON } = render(
      <FavoriteButton isSelected={true} onPress={mockOnPress} />
    );
    expect(toJSON()).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const { getByTestId } = render(
      <FavoriteButton isSelected={false} onPress={mockOnPress} />
    );
    
    const button = getByTestId('favorite-button');
    fireEvent.press(button);
    
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('does not call onPress when disabled', () => {
    const { getByTestId } = render(
      <FavoriteButton isSelected={false} onPress={mockOnPress} disabled={true} />
    );
    
    const button = getByTestId('favorite-button');
    fireEvent.press(button);
    
    expect(mockOnPress).not.toHaveBeenCalled();
  });

  it('applies custom size', () => {
    const component = render(
      <FavoriteButton isSelected={false} onPress={mockOnPress} size={32} />
    );
    expect(component).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(
      <FavoriteButton isSelected={false} onPress={mockOnPress} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

