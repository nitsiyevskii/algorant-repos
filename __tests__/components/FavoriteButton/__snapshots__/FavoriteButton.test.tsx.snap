// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F2F2F7",
      "borderRadius": 20,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 40,
    }
  }
  testID="favorite-button"
>
  <Text
    color="#000000"
    name="heart-outline"
    size={24}
    testID="icon"
  >
    heart-outline
  </Text>
</View>
`;
